crates/prompt_gateway/src/prompt_stream_context.rs

```diff
@@ -164,6 +165,7 @@ impl PromptStreamContext {
         let embedding_response: CreateEmbeddingResponse = match serde_json::from_slice(&body) {
             Ok(embedding_response) => embedding_response,
             Err(e) => {
+                debug!("error deserializing embedding response: {}", e);
```

Comment:
All of these debug logs should not be necessary. The error gets passed back to the user. Although if you do want debug logs printed to the console I would centralize that in send_server_error and simply print out the Display representation of the error struct.

Reply:
Reason I put these logs out is so I can see the source of log. If logs are coming out from one function e.g. send_server_error then it would be quite difficult to find out the source of problem

Reply:
The message should give you the source of the error.